Initial player setup

joshualeecoss opened the pull request:
- Set up basic state machine script
- Set up player Idle State and Move State scripts
- Set up player prefab
- Installed and set up Input Manager
- Set up basic player input
- Set up player controller script
- Set up player stats script

joshualeecoss committed:
Initial Commit

joshualeecoss committed:
State Machine, Basic Player Setup

- Added state machine logic
- Added basic player setup

joshualeecoss committed:
Player Prefab, Implemented Input System

- Created player prefab
- Refactored movement system to use the new Input System

joshualeecoss committed:
Player Stats, Updated Input, Proper Comments

- Added PlayerStats.cs with basic stats
- Updated movement to work with Unity's Input Manager
- Added proper comments so code is easier to read

josephlatina commented on Assets/Scenes/PlayerSetupScene.unity:
Are you able to add on to the `MainGameScene` instead of creating a separate scene? There's an object called CinemachineTargetGroup within that scene that you can attach a script that enables the main camera to follow the player as it moves around the terrain

You just need to merge in changes from master to your branch first

josephlatina replied:
Otherwise, it looks good and the capsule player reacts to the input controls!

joshualeecoss replied:
I had the separate scene set up for testing purposes, but I have now added the player to the main scene instead and deleted the testing scene.

josephlatina replied:
looks good!

josephlatina commented on Assets/Prefabs/Player/Player.prefab:
```diff
@@ -0,0 +1,188 @@
+  - component: {fileID: 6984755896152334956}
+  - component: {fileID: 6984755896152334959}
+  - component: {fileID: 6443449700755736254}
+  m_Layer: 0
```
Also, I had created a layer for 'Player', just need to set this player prefab to the 'Player' layer

joshualeecoss replied:
Player prefab is now set to the "Player" layer

joshualeecoss committed:
Merge branch 'main' into initial-player-setup-

joshualeecoss committed:
Added Player to main scene

- Added player to main game scene and attached to CinemachineTargetGroup
- Added player layer to player prefab

josephlatina commented:
![](https://github.trello.services/images/mini-trello-icon.png) [MS2-1: Initial Player Setup](https://trello.com/c/4APn7fxt/11-ms2-1-initial-player-setup)

josephlatina commented:
looks good!

jaizon2000 reviewed (approved):
Tried out the play controls and looks good!
